psl-core/src/main/java/org/linqs/psl/reasoner/sgd/SGDReasoner.java

```diff
@@ -56,64 +57,78 @@ public double optimize(TermStore baseTermStore) {
         }
 
         @SuppressWarnings("unchecked")
-        VariableTermStore<SGDObjectiveTerm, RandomVariableAtom> termStore = (VariableTermStore<SGDObjectiveTerm, RandomVariableAtom>)baseTermStore;
+        VariableTermStore<SGDObjectiveTerm, GroundAtom> termStore = (VariableTermStore<SGDObjectiveTerm, GroundAtom>)baseTermStore;
 
         termStore.initForOptimization();
 
-        double objective = -1.0;
+        long termCount = 0;
+        float movement = 0.0f;
+        double change = 0.0;
+        double objective = 0.0;
         double oldObjective = Double.POSITIVE_INFINITY;
+        float[] oldVariableValues = null;
```

Comment:
Add in a couple sentence commend here on why we keep around the old variables.
You can absorb the comment on lines 83-84 into this if you want.